// Contains API declaration for board settings bus.

syntax = "proto3";

package light_detector;

// Represents all the available settings options, which can be modified via settings bus.
enum SettingsType {
    Enable = 0;
    Disable = 1;
    Reset = 2;
    SetGain = 3;
    SetIntegralTime = 4;
}

// Represents settings bus request send from the client to the board.
message SettingsBusRequest {
    SettingsType settingsType = 1;
}

// Represents settings bus response send to the client from the board.
message SettingsBusResponse {
    uint32 deviceId = 1;
    SettingsType settingsType = 2;
    bool result = 3;
    uint32 nonce = 4;
}